### PR TITLE
implement missing tests for svid/jwtsvid/

### PR DESCRIPTION
**Pull request by willallves**

### CODE REVIEW

File: svid/jwtsvid/tests/check_svid.c [hunk 1]
```diff
@@ -57,7 +57,7 @@ START_TEST(test_jwtsvid_ParseInsecure)
     // jwtbundle_Bundle *bundle = jwtbundle_Load(td,
     // "./resources/jwk_keys.json", &err);
 
-    char token[]
+    char token1[]
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:60]: For each declared token, I'd use a more meaningful variable name, like token_with_audience_x or invalid_token, etc.

File: svid/jwtsvid/tests/check_svid.c [hunk 2]
```diff
@@ -144,6 +180,136 @@ START_TEST(test_jwtsvid_ParseAndValidate)
...
+    ck_assert_int_ge(shgeti(svid->claims, "iat"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "exp"), 0);
+    ck_assert_int_eq(svid->expiry, 1620000000);
+    ck_assert_ptr_ne(svid->id.path, NULL);
+    ck_assert_str_eq(svid->id.path, "/workload1");
+    ck_assert_ptr_ne(svid->id.td.name, NULL);
+    ck_assert_str_eq(svid->id.td.name, "example.com");
+    ck_assert_ptr_ne(svid->token, NULL);
+    ck_assert_str_eq(svid->token, token);
+
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:270]: This block of code looks like a copy of the code from lines 210 to 240. Couldn't refactor to a function and reuse it?
If you do so, please use a meaningful name for the function.

File: svid/jwtsvid/tests/check_svid.c [hunk 3]
```diff
@@ -144,6 +180,136 @@ START_TEST(test_jwtsvid_ParseAndValidate)
...
+    ck_assert_str_eq(marshal, token);
+
+    jwtbundle_Source_Free(source);
+    EVP_PKEY_free(pkey);
+    jwtsvid_SVID_Free(svid);
+}
+END_TEST
+
+START_TEST(test_jwtsvid_error5)
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:281]: Please add some comments to this test cases.

**Review by tjamir — CHANGES_REQUESTED** [review]
Some suggestion and a missing comment. Feel free to add any questions.

File: svid/jwtsvid/tests/check_svid.c [hunk 4]
```diff
@@ -278,7 +203,8 @@ START_TEST(test_jwtsvid_Marshal)
 }
 END_TEST
 
-START_TEST(test_jwtsvid_error5)
+// precondition: invalid jwt token with invalid signature
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:206]: I would recommend to add infortation about what is expected in this test (postcondition).
Like an error or success

File: svid/jwtsvid/tests/check_svid.c [hunk 5]
```diff
@@ -310,6 +236,62 @@ START_TEST(test_jwtsvid_error5)
 }
 END_TEST
 
+// precondition: invalid jwt token with subject not spifeeid
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:239]: I would recommend to add infortation about what is expected in this test (postcondition).
Like an error or success

File: svid/jwtsvid/tests/check_svid.c [hunk 6]
```diff
@@ -310,6 +236,62 @@ START_TEST(test_jwtsvid_error5)
...
+    char token[]
+        = "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImZmM2M1Yzk2LTM5MmUtNDZlZ"
+          "i1hODM5LTZmZjE2MDI3YWY3OCJ9."
+          "eyJzdWIiOiJzcGlmZmU6Ly9leGFtcGxlLmNvbS93b3JrbG9hZDEiLCJuYW1lIjoiSm9ob"
+          "iBEb2UiLCJpYXQiOjE1MTYyMzkwMjJ9."
+          "tIyeP2Alj6OE1KN3sLGDA89EdkTPqoHMi1hMyARyJZ_wpOpL57dgeeMSXWjbHYLW6tKIT"
+          "iOA9pcQVrCQRbcY9A";
+
+    err_t err;
+    jwtsvid_SVID *svid = jwtsvid_ParseInsecure(token, NULL, &err);
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:269]: If this jwt is invalid, shouldn't an error be expected and checked?

File: svid/jwtsvid/tests/check_svid.c [hunk 7]
```diff
@@ -310,6 +236,62 @@ START_TEST(test_jwtsvid_error5)
...
+    char token[]
+        = "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImZmM2M1Yzk2LTM5MmUtNDZlZ"
+        "i1hODM5LTZmZjE2MDI3YWY3OCJ9."
+        "eyJzdWIiOiJzcGlmZmU6Ly9leGFtcGxlLmNvbS93b3JrbG9hZDEiLCJuYW1lIjoiSm9obiB"
+        "Eb2UiLCJpc3MiOiJIUEUiLCJpYXQiOjE1MTYyMzkwMjIsImp0aSI6IkFCQ0RFRkdIIiwibmJ"
+        "mIjoxNDAwMDAwMDAwLCJhdWQiOlsiYXVkMSIsImF1ZDIiXX0.afcc79kL3JYqwrh1ABQe7og"
+        "Cy-WxWGb-G2YdmIKIeej3RMT-usawYCC8c1OH30Fi-_ygfzk_qLJnOs1t30tcEQ";
+
+    err_t err;
+    jwtsvid_SVID *svid = jwtsvid_ParseInsecure(token, NULL, &err);
```
Comment (tjamir) [inline svid/jwtsvid/tests/check_svid.c:288]: Check for error

**Review by tjamir — CHANGES_REQUESTED** [review]

File: svid/jwtsvid/tests/check_svid.c (hunk 6 above)
Comment (willallves) [reply svid/jwtsvid/tests/check_svid.c:276]: Added checked error, we will define documentation for errors.

File: svid/jwtsvid/tests/check_svid.c (hunk 7 above)
Comment (willallves) [reply svid/jwtsvid/tests/check_svid.c:298]: Added checked error, we will define documentation for errors.

File: svid/jwtsvid/tests/check_svid.c [hunk 8]
```diff
@@ -122,28 +126,183 @@ START_TEST(test_jwtsvid_ParseAndValidate)
...
+    ck_assert_ptr_ne(f, NULL);
+    EVP_PKEY *pkey = PEM_read_PrivateKey(f, NULL, NULL, NULL);
+    fclose(f);
+
+    err_t err = jwtbundle_Bundle_AddJWTAuthority(
+        bundle, "ff3c5c96-392e-46ef-a839-6ff16027af78", pkey);
+
+    ck_assert_uint_eq(err, NO_ERROR);
+
+    char token[]
```
Comment (otaviolcs3) [inline svid/jwtsvid/tests/check_svid.c:191]: Aplicar code-style pra esse arquivo.

**Review by otaviolcs3 — APPROVED** [review]

File: svid/jwtsvid/tests/check_svid.c [hunk 9]
```diff
@@ -4,6 +4,28 @@
...
+    ck_assert_int_ge(shgeti(svid->claims, "iat"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "exp"), 0);
+    ck_assert_int_eq(svid->expiry, 1620000000);
+    ck_assert_ptr_ne(svid->id.path, NULL);
+    ck_assert_str_eq(svid->id.path, "/workload1");
+    ck_assert_ptr_ne(svid->id.td.name, NULL);
+    ck_assert_str_eq(svid->id.td.name, "example.com");
+    ck_assert_ptr_ne(svid->token, NULL);
+    ck_assert_str_eq(svid->token, token);
+}
```
[the remark this answers is not in the document]
Comment (willallves) [reply svid/jwtsvid/tests/check_svid.c:26]: Nice!

**Pull request merged**